### PR TITLE
Internal sparse configs for segment

### PR DESCRIPTION
**Pull request by IvanPleshkov**
Required for https://github.com/qdrant/qdrant/pull/3154

Currently, `SparseVectorConfig` is used for public API and for `SegmentConfig` both. It's a bad implementation because the internals may be different. And https://github.com/qdrant/qdrant/pull/3154 is a example, where public API and internal configs have to be different. By sparse vectors public API, segment cannot decide, it segment appendable or not.

This PR introduces public REST API and internal segment config as separate entities. And also provides appendable flag for segment.

List of changes:
- Public API structure `SparseSegmentConfig` was renamed into `SparseVectorParams` and moved into collection crate. All fields are the same
- Public API structure `SparseIndexConfig`was renamed into `SparseIndexParams` and moved into collection crate. All fields are the same
- Internals `SparseVectorConfig` and `SparseIndexConfig` was created and added into `SegmentConfig`. `SparseIndexConfig` has state `SparseIndexState` wich can be mutable, immutable and mmap.
- Optimizers set sparse vector index as mmap or as immutable before optimizing

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by agourlay** [comment]
I will regen Python REST to check if there are any issues

**Comment by agourlay** [comment]
Please rename gRPC sparse index config to sparse index params to reflect REST.

File: lib/collection/src/config.rs [hunk 1]
```diff
@@ -357,12 +357,13 @@ impl CollectionParams {
                     (
                         name.into(),
                         SparseVectorDataConfig {
-                            index: Some(SparseIndexConfig {
+                            index: SparseIndexConfig {
                                 full_scan_threshold: params
```
Comment (agourlay) [inline lib/collection/src/config.rs:361]: Why not Optional?

Comment (IvanPleshkov) [reply lib/collection/src/config.rs:361]: Fixed

**Comment by IvanPleshkov** [comment]
> Please rename gRPC sparse index config to sparse index params to reflect REST.

Fixed

**Pull request merged**